[FEATURE] Add allowDropdown option

evanlouden opened the pull request.

dcyriller commented on addon/components/phone-input.js:
```diff
@@ -44,6 +46,17 @@ export default Component.extend({
+      @type {boolean}
+    */
+
+    this.allowDropdown = isPresent(this.allowDropdown)
```
Could we make it an opt-in instead? :pray:

If we make it an opt-in, we won't need a major release (it would be breaking). Also, we would remain inline with [intl-tel-input](https://github.com/jackocnr/intl-tel-input#options) where the default is true.

dcyriller commented on tests/dummy/app/pods/docs/components/phone-input/all-options/template.hbs:
```diff
@@ -31,3 +31,16 @@
+{{#docs-demo as |demo|}}
+  {{#demo.example name="phone-input-allow-dropdown-option.hbs"}}
+
+    <p>{{phone-input allowDropdown=false initialCountry="fr" update=(action "handleUpdate")}}</p>
```
```suggestion
    <p>{{phone-input allowDropdown=false number=allowDropdownNumber initialCountry="fr" update=(action "updateAllowDropdown")}}</p>
```
If we add a number and a dedicated update method, it won't interfere with other phone-input occurences on this page.

Note that we probably want to update the `tests/dummy/app/pods/docs/components/phone-input/all-options/component.js` to implement the update method. That way, the sandbox UI will be 100% working.

evanlouden commented:
Really sorry I've left this here for so long. I was confident my company was going to need this feature when we added this add on. Turns out we didn't. And around that time muted my github notifications. 😝  I've updated the PR and happy to get it over the finish line. Otherwise, feel free to close it. 😄

dcyriller commented on tests/integration/components/phone-input-test.js:
```diff
@@ -128,4 +128,20 @@ module('Integration | Component | phone-input', function(hooks) {
+  test('can prevent the dropdown', async function(assert) {
+    assert.expect(1);
+
+    await this.owner.lookup('service:phone-input').load();
```
The library is loaded in a `beforeEach` hook: https://github.com/qonto/ember-phone-input/pull/134/files#diff-0e2859b8508db13adaa5918464d0aeecR10

You can safely remove this line.

dcyriller commented on tests/integration/components/phone-input-test.js:
```diff
@@ -128,4 +128,20 @@ module('Integration | Component | phone-input', function(hooks) {
+    await this.owner.lookup('service:phone-input').load();
+
+    this.set('update', value => {
+      this.set('allowDropdown', value);
```
Sorry I was not clear enough on the update method. Here you need to update the number, not the allowDropdown option.

dcyriller replied:
Or you might be able to remove it at all

dcyriller commented:
Released in a v3.1.0

evanlouden commented:
@dcyriller Thanks for your help!
